### PR TITLE
Delete the cluster at correct index to prevent intermittent test failures

### PR DESCRIPTION
**Pull request by sankarpn**
The test was deleting a different cluster than intended and it was causing intermittent failures. The changes in this PR adds a new utility method to delete cluster by determining the correct index and then patching to remove it from the domain resource.

https://build.weblogick8s.org:8443/job/weblogic-kubernetes-operator-kind-new/16782/

### CODE REVIEW

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/DomainUtils.java [hunk 1]
```diff
@@ -407,6 +410,46 @@ public static void deleteDomainResource(String domainNS, String domainUid) {
...
+   * @param domainUid name of the domain resource
+   * @param namespace namespace in which domain resource exists
+   * @throws ApiException throws when cluster interaction fails
+   */
+  public static void removeClusterInDomainResource(String clusterName, String domainUid, String namespace)
+      throws ApiException {
+    LoggingFacade logger = getLogger();
+    logger.info("Removing the cluster {0} from domain resource {1}", clusterName, domainUid);
+    DomainResource domainCustomResource = getDomainCustomResource(domainUid, namespace);
+    logger.info(Yaml.dump(domainCustomResource));
```
Comment (doxiao) [inline integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/DomainUtils.java:426]: This may be too verbose.

**Review by doxiao — APPROVED** [review]
LGTM.

**Review by ankedia — APPROVED** [review]
LGTM, I just have a minor comment.

**Review by xiancao — APPROVED** [review]

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/DomainUtils.java (hunk 1 above)
Comment (sankarpn) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/utils/DomainUtils.java]: removed the resource dump

File: integration-tests/src/test/java/oracle/weblogic/kubernetes/ItKubernetesDomainEvents.java [hunk 2]
```diff
@@ -479,6 +477,7 @@ void testScaleDomainAndVerifyCompletedEvent() {
     scaleDomainAndVerifyCompletedEvent(1, ScaleAction.scaleDown, true, domainNamespace4);
     scaleDomainAndVerifyCompletedEvent(2, ScaleAction.scaleUp, true, domainNamespace4);
     shutdownDomain(domainUid, domainNamespace4);
```
[the remark this answers is not in the document]
Comment (sankarpn) [reply integration-tests/src/test/java/oracle/weblogic/kubernetes/ItKubernetesDomainEvents.java]: removed the shutdown call

**Pull request merged**